### ISSUE TITLE
Cannot modify the configuration of a local custom cop

### ISSUE THREAD

**Issue by r7kamura** [rubocop/rubocop#14364]
## Expected behavior

I defined a custom cop in `lib/rubocop/cop/foo/bar.rb` and attempted to change its configuration in `.rubocop.yml`.

For example, if I set `SafeAutoCorrect: false` for this custom cop, I expect `.rubocop_todo.yml` to include a comment like "This cop supports unsafe autocorrection" for this cop.

```yaml
require:
  - ./lib/rubocop/cop/foo/bar

Foo/Bar:
  SafeAutoCorrect: false
```

## Actual behavior

However, when I actually ran RuboCop, it seemed that this configuration wasn't applied, and the cop was still commented as supporting **safe** autocorrection.

```yaml
# Offense count: 1
# This cop supports safe autocorrection (--autocorrect).
Foo/Bar:
  Exclude:
    - 'example.rb'
```

Is this the intended behavior? Or is defining a custom cop locally and configuring it directly in `.rubocop.yml` an unsupported operation that shouldn’t be done, and in such cases, should it instead be packaged and loaded as a plugin?

## Steps to reproduce the problem

I've prepared a repository that reproduces the behavior described above. It contains a very simple custom cop, its configuration added in `.rubocop.yml`, and a generated `.rubocop_todo.yml`.

- https://github.com/r7kamura/rubocop_custom_cop_example

## RuboCop version

```
$ bundle exec rubocop -V
1.78.0 (using Parser 3.3.8.0, rubocop-ast 1.45.1, analyzing as Ruby 2.7, running on ruby 3.4.2) [x86_64-linux]
```

**Comment by Earlopain** [comment]
Yes, I think this should be supported. If there are offenses for a cop that is not found in the default config, looking in top-level `.rubocop.yml` makes sense:

```diff
diff --git a/lib/rubocop/formatter/disabled_config_formatter.rb b/lib/rubocop/formatter/disabled_config_formatter.rb
index b105a1dc9..c6f18ab53 100644
--- a/lib/rubocop/formatter/disabled_config_formatter.rb
+++ b/lib/rubocop/formatter/disabled_config_formatter.rb
@@ -154,7 +154,7 @@ module RuboCop
         output_buffer.puts "# Offense count: #{offense_count}" if show_offense_counts?
 
         cop_class = Cop::Registry.global.find_by_cop_name(cop_name)
-        default_cfg = default_config(cop_name)
+        default_cfg = default_config(cop_name) || @config_for_pwd[cop_name]
 
         if supports_safe_autocorrect?(cop_class, default_cfg)
           output_buffer.puts '# This cop supports safe autocorrection (--autocorrect).'
```

However there are a handful of tests that are too specific. I would invite you to open a PR to fix this

**Comment by RedZapdos123** [comment]
Opened a PR for this fix.
